scripts/installDependencies.sh

```diff
@@ -0,0 +1,136 @@
+{
+    local version=${1:-"v1.15.1"}
+    local opmPrefix="opmup"
+    local platform=""
+    local architecture=""
+
+    # translate OSTYPE to expected platform name
+    if [[ "${OSTYPE}" == "linux-gnu"* ]]; then
+        platform="linux"
+    elif [[ "${OSTYPE}" == "darwin"* ]]; then
+        platform="darwin"
+    else
+        # unsupported os
+        return 1
+    fi
+
+    # translate HOSTTYPE to expected architecture name
+    if [[ "${HOSTTYPE}" == "x86_64" ]]; then
+        architecture="amd64"
+    elif [[ "${HOSTTYPE}" == "s390x" ]]; then
+        architecture="s390x"
+    elif [[ "${HOSTTYPE}" == "powerpc64le" ]]; then
+        architecture="ppc64le-opm"
+    else
+        # unsupported architecture
+        return 1
+    fi
+    
+    # download the upstream opm taking into account the version, platform, and architecture we're running on
+    assetURL=$(curl -s "https://api.github.com/repos/operator-framework/operator-registry/releases/tags/${version}" | jq --argjson platform "\"${platform}\"" --argjson architecture "\"${architecture}\"" -r '.assets[] | select(.browser_download_url | contains( ($platform + "-" + $architecture) ) ) | .url')
+    if ! "${TOOLS_BIN}/${opmPrefix}${version}" version 2>/dev/null && [[ -n ${assetURL} ]] ; then
+        curl -sL -H "Accept: application/octet-stream" -o "${TOOLS_BIN}/${opmPrefix}${version}" "${assetURL}" && chmod 755 "${TOOLS_BIN}/${opmPrefix}${version}"
+    fi
+
+}
+
+# installs downstream opm which requires docker for image extraction
+function installDownstreamOpm() 
+{
+    local opmCatalogBinaryImage="${1:-"registry.redhat.io/openshift4/ose-operator-registry:v4.6.0"}"
+    local version="${2:-"v1.14.3"}"
+    local opmPrefix="opmdown"
+    local executableName=""
+
+    if [[ "${OSTYPE}" == "linux-gnu"* ]]; then
+        executableName="opm"
+    elif [[ "${OSTYPE}" == "darwin"* ]]; then
+        executableName="darwin-amd64-opm"
+    else
+        # unsupported os
+        return 1
+    fi
+
+    # check if we already have it
+    if ! "${TOOLS_BIN}/${opmPrefix}${version}" version 2>/dev/null; then
+        # docker login to redhat registry
+        if echo "${RH_PASSWORD}" | docker login registry.redhat.io -u "\"${RH_USER}\"" --password-stdin &>/dev/null; then
+            # create a offline container
+            if containerReference=$(docker create "${opmCatalogBinaryImage}"); then
+                # extract binary to tools bin and make it executable
+                docker cp --follow-link "${containerReference}:/usr/bin/${executableName}" "${TOOLS_BIN}" && \
+                    mv "${TOOLS_BIN}/${executableName}" "${TOOLS_BIN}/${opmPrefix}${version}" && \
+                    chmod 755 "${TOOLS_BIN}/${opmPrefix}${version}"
+            else
+                # can't create offline container
+                return 1
+            fi
+        else
+            # can't login
+            return 1
+        fi
+    fi
+}
+
+# installs the "oc" executable
+function installOC() 
```

Comment:
`oc` is a openshift-specific tool and we don't necessarily want to depend on it in our OLM upstream repository

Reply:
Our usage of this tool is currently non-existent... I had it in here since some of our existing bash based test scripts used oc and thought we might need it. Using kubectl (if necessary at some point) is probably sufficient. @cdjohnson might have ideas on why this might be needed.

Reply:
`kubectl` is sufficient.  `oc` is a legacy artifact of my tool and has no dependencies on oc itself.

Reply:
Thanks for clarification Chris. I'll pull the installOC portions out then.